### DISCUSSION TITLE
Unexpected behaviour with properties which contain Eloquent collections

### DISCUSSION THREAD

**Discussion by Forsakenrox** [livewire/livewire#5282] [1. Help]
Hi, im working on a very highloaded project which require a lot of optimisations, and for some sections i select this library to little speed up myself. But i did not expect what im meet with unexpected behavior of my livewire component. Let me explain.

to reproduce this behavior i create a very simple component with 2 different public properties:
```
<?php

namespace App\Http\Livewire;

use App\Models\Post;
use Livewire\Component;

class TestComponent extends Component
{
    public bool $isModalOpenned = false;
    public $posts;

    public function mount()
    {
        $this->getPosts();
    }

    public function getPosts()
    {
        $this->posts = Post::get();
    }

    public function render()
    {
        return view('livewire.test-component');
    }
}

```
Simple, right ? 
For optimisation purposes i use special method `getPosts()` to prevent updating on every request, and update property `$posts` only when it realy required.
Additional  i was make a property `$isModalOpenned`, just for example, to change modal window visibility (from `wire:click=$set('isModalOpenned', true`). And it work as well, apart from my expectation from queries to database.
![image](https://user-images.githubusercontent.com/11795286/193480403-6afdc2f9-6870-442e-8c40-25afa5fc68c7.png)

I expect which `$posts` property will updating ONLY on the `mount()` state of component, instead of this - that property updating on every request, and every changing of any other non database properties. 

Is this behaviour is correct for this library? Confirm or refute my statements. thank you very much.
And if that's normal behavior, does it mean that it's impossible to do anything high performance on livewire?

**Comment by ousid** [comment]
Hey, @Forsakenrox 
I didn't face this issue before, but what about [Defer Loading?](https://laravel-livewire.com/docs/2.x/defer-loading)

**Reply by Forsakenrox** [reply]
Hey, this is a little different thing. My problem is not related to component initialization. It works well. The problem shows up when I make a second, and etc requests. The properties associated with the eloquent collection update themselves on every interaction with the livewire component, do querying the database. But it **shouldn't** be happened.

**Comment by joshhanley** [comment]
@Forsakenrox if you store eloquent models or an eloquent collection of models in a public property, for security reasons, Livewire only sends the class name and list of IDs to the front end. So to restore the Eloquent Collection, Livewire has to make a DB request with the specified IDs to repopulate the property.

To get around this, I would not store your posts directly in the property and instead convert any required data to an array and only store what data you need.

Hope this helps!

**Reply by Forsakenrox** [reply]
thanks for attention, i already solve this issue with computed property with cached result with loosing of server side rendering, but i take a HUGE performance improvement. Using arrays in my case prohibited because i have a lot of libraries methods on the models.

**Reply by joshhanley** [reply]
@Forsakenrox ah yeah that makes sense. computed properties with cache is also a great solution! 😁

**Reply by joshhanley** [reply]
Plus cache is more secure too than sending all that data to the front end!
